### PR TITLE
refactor(EditTearsheetForm): add Typescript types

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
Closes [#4283](https://github.com/carbon-design-system/ibm-products/issues/4283)

This PR introduces Typescript types to EditTearsheetForm

#### What did you change?
Changed the file and component to using Typescript

#### How did you test and verify your work?
Ran tests and build locally

### CODE REVIEW

File: packages/ibm-products/src/components/EditTearsheet/EditTearsheetForm.tsx [hunk 1]
```diff
@@ -37,10 +84,10 @@ export let EditTearsheetForm = forwardRef(
 
       // Collect any other property values passed in.
       ...rest
-    },
-    ref
+    }: EditTearsheetFormProps,
+    ref: ForwardedRef<HTMLDivElement>
   ) => {
-    const formContext = useContext(FormContext);
+    const formContext = useContext(FormContext) as any;
```
Comment (matthewgallo) [inline packages/ibm-products/src/components/EditTearsheet/EditTearsheetForm.tsx:90]: You could avoid using `any` here if you provide a type that is shared between when this context is created in `EditTearsheet` and then here as well.
```
export const FormContext = createContext<FormContextType | null>(null);
```

```
const formContext = useContext(FormContext) as FormContextType;
```

where `FormContextType` could be something like
```
type FormContextType = {
  currentForm: number;
  setFormTitle: () => void;
};
```

File: packages/ibm-products/src/components/EditTearsheet/EditTearsheetForm.tsx [hunk 2]
```diff
@@ -22,6 +22,53 @@ const defaults = {
...
+   * This optional prop will render your form content inside of a fieldset html element
+   * and is defaulted to true.
+   * You can set this prop to `false` if you have multiple fieldset elements or want to control the children of your Full Page's form content.
+   */
+  hasFieldset?: boolean;
+
+  /**
+   * Sets an optional subtitle on the form component
+   */
+  subtitle: string;
```
Comment (matthewgallo) [inline packages/ibm-products/src/components/EditTearsheet/EditTearsheetForm.tsx:51]: This should be optional
```suggestion
  subtitle?: string;
```

**Review by amal-k-joy — CHANGES_REQUESTED** [review]

File: packages/ibm-products/src/components/EditTearsheet/EditTearsheet.js [hunk 3]
```diff
@@ -19,7 +19,7 @@ const blockClass = `${pkg.prefix}--tearsheet-edit`;
 // This is a general context for the forms container
 // containing information about the state of the container
 // and providing some callback methods for forms to use
-export const FormContext = createContext(null);
+export const FormContext = (createContext < FormContext) | (null > null);
```
[the remark this answers is not in the document]
Comment (IgnacioBecerra) [reply packages/ibm-products/src/components/EditTearsheet/EditTearsheet.js:22]: Hey Amal, you're right, but I'd like to export the FormContextType from EditTearsheet.tsx, which will be introduced in #5082, wondering if we can get that in first?

[the remark this answers is not in the document]
Comment (IgnacioBecerra) [reply packages/ibm-products/src/components/EditTearsheet/EditTearsheet.js]: Should be fixed now!

**Review by amal-k-joy — APPROVED** [review]

**Review by matthewgallo — APPROVED** [review]

**Pull request merged**